bitcash/network/APIs/BitcoinDotComAPI.py

```diff
@@ -0,0 +1,132 @@
+import requests
+from decimal import Decimal
+from bitcash.exceptions import InvalidEndpointURLProvided
+from bitcash.network import currency_to_satoshi
+from bitcash.network.meta import Unspent
+from bitcash.network.transaction import Transaction, TxPart
+
+# This class is the interface for Bitcash to interact with
+# Bitcoin.com based RESTful interfaces.
+
+BCH_TO_SAT_MULTIPLIER = 100000000
+# TODO: Refactor constant above into a 'constants.py' file
+
+
+class BitcoinDotComAPI:
+    """ rest.bitcoin.com API """
+
+    def __init__(self, network_endpoint: str):
+        try:
+            assert isinstance(network_endpoint, str)
```

Comment:
Just to note, if anyone uses Python optimization `-O`, this will get skipped.

Reply:
You mean it does so for asserts in the code? That's interesting, wasn't aware of that.

Reply:
I learned it just yesterday. I've also never used Python's optimized mode before and am not sure how often anyone does.

Reply:
Fair enough. It's something I'll keep in mind in the future. These asserts are really more for convenience though, as an invalid URL will still fail later down the road anyway (if someone does have the `-O` flag set).